lib/iris/analysis/stats.py

```diff
@@ -22,108 +22,45 @@
 
 import numpy as np
-
 import iris
+from iris.util import broadcast_to_shape
 
 
-def _get_calc_view(cube_a, cube_b, corr_coords):
-    """
-    This function takes two cubes and returns cubes which are
-    flattened so that efficient comparisons can be performed
-    between the two.
-
-    Args:
-
-    * cube_a:
-        First cube of data
-    * cube_b:
-        Second cube of data, compatible with cube_a
-    * corr_coords:
-        Names of the dimension coordinates over which
-        to calculate correlations.
-
-    Returns:
-
-    * reshaped_a/reshaped_b:
-        The data arrays of cube_a/cube_b reshaped
-        so that the dimensions to be compared are
-        flattened into the 0th dimension of the
-        return array and other dimensions are
-        preserved.
-    * res_ind:
-        The indices of the dimensions that we
-        are not comparing, in terms of cube_a/cube_b
-
+def pearsonr(cube_a, cube_b, corr_coords=None, weights=None, mdtol=1.):
     """
+    Calculates the n-D Pearson's r correlation cube over the dimensions
+    associated with the given coordinates.
 
-    # Following lists to be filled with:
-    # indices of dimension we are not comparing
-    res_ind = []
-    # indices of dimensions we are comparing
-    slice_ind = []
-    for i, c in enumerate(cube_a.dim_coords):
-        if not c.name() in corr_coords:
-            res_ind.append(i)
-        else:
-            slice_ind.append(i)
-
-    # sanitise input
-    dim_coord_names = [c.name() for c in cube_a.dim_coords]
-    if corr_coords is None:
-        corr_coords = dim_coord_names
-
-    if ([c.name() for c in cube_a.dim_coords] !=
-            [c.name() for c in cube_b.dim_coords]):
-        raise ValueError("Cubes are incompatible.")
-
-    for c in corr_coords:
-        if c not in dim_coord_names:
-            raise ValueError("%s coord "
-                             "does not exist in cube." % c)
-
-    # Reshape data to be data to correlate in 0th dim and
-    # other grid points in 1st dim.
-    # Transpose to group the correlation data dims before the
-    # grid point dims.
-    data_a = cube_a.data.view()
-    data_b = cube_b.data.view()
-    dim_i_len = np.prod(np.array(cube_a.shape)[slice_ind])
-    dim_j_len = np.prod(np.array(cube_a.shape)[res_ind])
-    reshaped_a = data_a.transpose(slice_ind+res_ind)\
-                       .reshape(dim_i_len, dim_j_len)
-    reshaped_b = data_b.transpose(slice_ind+res_ind)\
-                       .reshape(dim_i_len, dim_j_len)
-
-    return reshaped_a, reshaped_b, res_ind
-
-
-def pearsonr(cube_a, cube_b, corr_coords=None):
-    """
-    Calculates the n-D Pearson's r correlation
-    cube over the dimensions associated with the
-    given coordinates.
-
-    Returns a cube of the correlation between the two
-    cubes along the dimensions of the given
-    coordinates, at each point in the remaining
+    Returns a cube of the correlation between the two cubes along the
```

Comment:
This description could go in the Returns section after the arguments, see below.